src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -207,23 +208,30 @@ private Resource createOverseasEntityTransactionResource(String submissionUri) {
     }
 
     private void updateTransactionWithLinksAndCompanyName(Transaction transaction,
-                                                          EntityNameDto entityNameDto,
+                                                          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
                                                           String submissionId,
                                                           String submissionUri,
                                                           Resource overseasEntityResource,
                                                           String loggingContext,
                                                           boolean addResumeLinkToTransaction) throws ServiceException {
         String entityName = null;
+        var entityNameDto = overseasEntitySubmissionDto.getEntityName();
         if (Objects.nonNull(entityNameDto)) {
             entityName = entityNameDto.getName();
         }
+
         transaction.setCompanyName(entityName);
 
         transaction.setResources(Collections.singletonMap(submissionUri, overseasEntityResource));
 
         if (addResumeLinkToTransaction) {
-            final var resumeJourneyUri = String.format(RESUME_JOURNEY_URI_PATTERN, transaction.getId(), submissionId);
-            transaction.setResumeJourneyUri(resumeJourneyUri);
+            if (overseasEntitySubmissionDto.isForUpdate()) {
+                final var updateResumeJourneyUri = String.format(UPDATE_RESUME_JOURNEY_URI_PATTERN, transaction.getId(), submissionId);
+                transaction.setResumeJourneyUri(updateResumeJourneyUri);
```

Comment:
Minor, but I would have made the call to `setResumeJourneyUri` outside of the conditional statement (to avoid the repeated line of code), after the correct URI has been set. Up to you.

Reply:
Done !